proto/auth.proto

```diff
@@ -69,98 +71,9 @@ message _GenerateApiTokenRequest {
         Expires expires = 2;
     }
 
-    // Aliases for categories of functionality.
-    enum CacheRole {
-      CachePermitNone = 0;
-      // Restricts access to apis that read and write data from caches: No higher level resource description or modification.
-      CacheReadWrite = 1;
-      // Restricts access to apis that read from caches: No higher level resource description or modification.
-      CacheReadOnly = 2;
-      // Doesn't allow conditional write APIs (SetIfNotExists, IncreaseTTL etc)
-      CacheWriteOnly = 3;
-    }
-
-    // Aliases for categories of functionality.
-    enum TopicRole {
-      TopicPermitNone = 0;
-      // Restricts access to apis that read and write data from topics: No higher level resource description or modification.
-      TopicReadWrite = 1;
-      // Restricts access to apis that read from topics: No higher level resource description or modification.
-      TopicReadOnly = 2;
-      // Only publish allowed
-      TopicWriteOnly = 3;
-    }
-
     string auth_token = 3;
 
-    enum SuperUserPermissions {
-      SuperUser = 0;
-    }
-
-    message Permissions {
-      oneof kind {
-        SuperUserPermissions super_user = 1;
-        ExplicitPermissions explicit = 2;
-      }
-    }
-
-    message ExplicitPermissions {
-      repeated PermissionsType permissions = 1;
-    }
-
-    message PermissionsType {
-      oneof kind {
-        CachePermissions cache_permissions = 1;
-        TopicPermissions topic_permissions = 2;
-      }
-
-      message All {}
-
-      message CacheSelector {
-        oneof kind {
-          string cache_name = 1;
-        }
-      }
-
-      message CacheItemSelector {
-        oneof kind {
-          bytes key = 1;
-          bytes key_prefix = 2;
-        }
-      }
-
-      message CachePermissions {
-        CacheRole role = 1;
-        oneof cache {
-          All all_caches = 2;
-          CacheSelector cache_selector = 3;
-        }
-        oneof cache_item {
-          All all_items = 4;
-          CacheItemSelector item_selector = 5;
-        }
-      }
-
-      message TopicSelector {
-        oneof kind {
-          string topic_name = 1;
-        }
-      }
-
-      message TopicPermissions {
-        TopicRole role = 1;
-        oneof cache {
-          All all_caches = 2;
-          CacheSelector cache_selector = 3;
-        }
-        oneof topic {
-          All all_topics = 4;
-          TopicSelector topic_selector = 5;
-        }        
-      }
-    }
-
-    Permissions permissions = 4;
+  permissions.Permissions permissions = 4;
```

Comment:
Can you try generating the protos for the web sdk and compiling the web sdk with them? we used some hacks there to get it to work with strictjs and I am worried that moving these messages to different packages is going to break it.

Reply:
@cprice404 FYI ^

Reply:
Got a successful build without `package permissions`, did have to change imports, here's the `javascript-web` changes
https://github.com/momentohq/client-sdk-javascript/pull/729

![image](https://github.com/momentohq/client-protos/assets/124546494/b7674874-6fa1-4380-b5e3-852961b13db9)

Reply:
Okay, this was a real pain. For `Javascript-web`, I removed the `package` declarations, but in doing so, rust builds will fail since tonic server seems to build protos into the out directory only if they're a package, couldn't find anything in [the docs to help with this](https://docs.rs/tonic-build/latest/tonic_build/struct.Builder.html#method.server_attribute). 

Noticing that the `javascript-web` build script actually comments out `package`, I thought I would keep the package declares and  update `permissions.Permissions` to `Permissions` for  `javascript-web` only. It ain't pretty.


Got it working and auth test running in `javascript-web`,  `javascript-nodejs` I was having much worse luck with, `@grpc/grpc-js` is shared between the packages, so `ChannelCredentials, Interceptor` had conflicts between the two different versions.
![image](https://github.com/momentohq/client-protos/assets/124546494/7f0ed545-92b7-4945-bfe8-cd8750b861a6)


Alternatively to avoid all of this, we could maintain two copies of the permissions object in `auth` and `token`, but not a hug fan of this.

Reply:
the javascript-web build is already a rube goldberg machine. Personally I am okay with pushing more horror and complexity there rather than letting it bleed into other languages/SDKs. If @krispraws is.

Reply:
dumb question - Do we need to put each proto file/each grpc service in its own package? Can tokens.proto, permissions.proto and auth.proto all live in the same `auth` package?

Reply:
I agree with @cprice404 - we can push the complexity into javascript-web builds.